packages/taiko-client/preconfapi/builder/blob.go

```diff
@@ -34,40 +35,101 @@ func NewBlobTransactionBuilder(
 // return an unsigned transaction, intended for preconfirmations.
 func (b *BlobTransactionBuilder) BuildUnsigned(
 	_ context.Context,
-	txListBytes []byte,
-	l1StateBlockNumber uint32,
-	timestamp uint64,
-	coinbase common.Address,
-	extraData [32]byte,
+	opts BuildUnsignedOpts,
 ) (*types.Transaction, error) {
-	compressedTxListBytes, err := utils.Compress(txListBytes)
-	if err != nil {
-		return nil, err
+	encodedParams := make([][]byte, 0)
+
+	blobs := make([]*eth.Blob, 0)
+
+	type blobInfo struct {
+		index  uint8
+		offset uint64
+		length uint64
 	}
 
-	var blob = &eth.Blob{}
-	if err := blob.FromData(compressedTxListBytes); err != nil {
-		return nil, err
+	blobInfos := make([]blobInfo, 0)
+
+	var totalBytes []byte
+
+	var idx uint8
+
+	for i, opt := range opts.BlockOpts {
+		txListBytes, err := signedTransactionsToTxListBytes(opt.SignedTransactions)
+		if err != nil {
+			return nil, err
+		}
+
+		compressedTxListBytes, err := utils.Compress(txListBytes)
+		if err != nil {
+			return nil, err
+		}
+
+		totalWithCurrentTxList := append(totalBytes, compressedTxListBytes...)
+
+		// if we have exceeded max blob data size, we can make a blob with all the data
+		// *except* this one.
+		if len(totalWithCurrentTxList) >= eth.MaxBlobDataSize {
+			var blob = &eth.Blob{}
+			if err := blob.FromData(totalBytes); err != nil {
+				return nil, err
+			}
+
+			blobs = append(blobs, blob)
+
+			// clear the bytes array and increment the blob index for next iterations of the loop
+			totalBytes = []byte{}
```

Comment:
Why clean `totalBytes` at there but not when `blob.FromData(totalBytes)` is called?

Reply:
we do, it's right above this

Reply:
Put them together be easier to understand？

Reply:
what?

Reply:
Use it and clean it in the same place.

Reply:
it's literally the previous line of code? 
![Screenshot_20240715-204258](https://github.com/user-attachments/assets/26d64dbd-bc6f-41b6-b543-5a374ab7c46c)

Reply:
Actually, Line 73 already exists one.